src/kOS/Suffixed/PartModuleField/PartModuleFields.cs

```diff
@@ -196,7 +196,7 @@ protected virtual ListValue AllFields(string formatter)
             {
                 returnValue.Add(new StringValue(string.Format(formatter,
                                               IsEditable(field) ? "settable" : "get-only",
-                                              field.guiName.ToLower(),
+                                              field.guiName.Length > 0 ? field.guiName.ToLower() : field.name.ToLower(),
```

Comment:
This seems to fix the problem only for the suffix ``:ALLFIELDS``.   But the name it returns still wouldn't be recognized in all the other places where you can use these names, like ``:ALLFIELDNAMES``, or ``:HASFIELD``, or ``:GETFIELD``, or ``:SETFIELD``, etc.

For those, they're still only looking at the ``guiName``.  So this would make the user think kOS will let them use the name when it won't.

To really fix the problem would require coming up with some kind of method that can be re-used in all the places the ``guiName`` is being used now, to perform this same "use guiName unless it's blank then use not-gui name." technique all over the place.

Reply:
I also wonder if the same logic needs to be applied not just to Field names but also to Event names (buttons on the menu) and Action group actions.

Reply:
Hmm, good point, I'll do a bit more testing on this. I've not seen it for actions and events, but that's not too say it couldn't happen.